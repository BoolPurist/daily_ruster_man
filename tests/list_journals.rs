mod common;
use daily_ruster_man::{
    core::{date_models::find_by::FindByYearMonthDay, app_options::AppOptions, list_queries},
    cli::app_args::GenerellArgs,
};

#[test]
fn should_list_all_daily_journals() {
    let files = common::create_sample_data_folder();
    let querry =
        FindByYearMonthDay::new(None, None, None).expect("Invalid date for querry provided");
    let general = GenerellArgs::new(
        false,
        None,
        Some(files.path().to_str().unwrap().to_string()),
    );

    let app_options = AppOptions::with(general);
    let all_daily_journals = list_queries::fetch_all_daily_names(&querry, &app_options)
        .expect("Could not fetch all daily journals");

    insta::assert_yaml_snapshot!(all_daily_journals);
}
