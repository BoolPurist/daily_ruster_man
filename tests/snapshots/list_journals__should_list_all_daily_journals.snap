---
source: tests/list_journals.rs
expression: all_daily_journals
---
- 2023 03 09
- 2023 03 08
- 2023 02 02
- 1999 02 21
- 1999 01 21
- 1988 11 22
- 1788 08 12

